hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieDefaultTimeline.java

```diff
@@ -490,6 +493,19 @@ public Option<HoodieInstant> getFirstNonSavepointCommit() {
     }
     return firstNonSavepointCommit;
   }
+
+  public Option<HoodieInstant> getLastClusterCommit() {
+    return  Option.fromJavaOptional(getCommitsTimeline().filter(s -> s.getAction().equalsIgnoreCase(HoodieTimeline.REPLACE_COMMIT_ACTION))
+        .getReverseOrderedInstants()
+        .filter(i -> {
+          try {
+            HoodieCommitMetadata metadata = TimelineUtils.getCommitMetadata(i, this);
```

Comment:
This introduced a regression that the pending clustering is no longer considered when figuring out the `lastClusteringInstant`.